src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Body/Body/Body.xtuml

```diff
@@ -966,6 +966,476 @@ return false;
+	'blockId',
+	"ba5eda7a-def5-0000-0000-000000000005",
+	0,
+	'',
+	"00000000-0000-0000-0000-000000000000",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("907fc361-22bf-479c-aca2-7cf65f0b1d86",
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	'declarationLineNumber',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"843648db-93dc-4d99-96f3-b71856f50a78",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("5bcb93a8-d6c6-4f4b-97b8-dc1a6a53e093",
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	'declarationLineColumn',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"907fc361-22bf-479c-aca2-7cf65f0b1d86",
+	'');
+INSERT INTO O_TFR
+	VALUES ("73a94ff8-3736-4fc5-b3ef-4d1b9823e200",
+	"d1c9af7b-9df6-432a-a25e-96cab24fc6af",
+	'findVariableLocationForVariable',
+	'',
+	"bc3722c8-5188-4529-8519-cf9eb9816b17",
+	1,
+	'select any variable from instances of V_VAR where (selected.Var_ID == param.var_id);
+select any firstLocation from instances of V_LOC
+                                    where (selected.Id == GD::NULL_UNIQUE_ID());
+foundMatchingVariable = false;
+select many locations related by variable->V_LOC[R835];
+for each location in locations
+  if(empty firstLocation)
+    firstLocation = location;
+  else
+    if(location.LineNumber < firstLocation.LineNumber)
+      firstLocation = location;
+    elif(location.LineNumber == firstLocation.LineNumber
+                       and location.StartPosition < firstLocation.StartPosition)
+      firstLocation = location;
+    end if;
+  end if;
+  if(location.LineNumber == param.declarationLineNumber
+                      and location.StartPosition == param.declarationLineColumn)
+    foundMatchingVariable = true;
+  end if;
+end for;
+if(not_empty firstLocation and not param.checkLocation)
+  return firstLocation.convertToInstance();
+end if;
+if(not_empty firstLocation and foundMatchingVariable)
+  return firstLocation.convertToInstance();
+end if;
+return ::emptyInstance();',
+	1,
+	'',
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	0);
+INSERT INTO O_TPARM
+	VALUES ("e98867dd-7cc8-41d6-8196-beb8abd57dde",
+	"73a94ff8-3736-4fc5-b3ef-4d1b9823e200",
+	'declarationLineNumber',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"00000000-0000-0000-0000-000000000000",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("89815f41-55ac-4195-9852-20c958dbe0b9",
+	"73a94ff8-3736-4fc5-b3ef-4d1b9823e200",
+	'declarationLineColumn',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"e98867dd-7cc8-41d6-8196-beb8abd57dde",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("d7f33178-5c0b-46b5-8969-f82b80b99e87",
+	"73a94ff8-3736-4fc5-b3ef-4d1b9823e200",
+	'var_id',
+	"ba5eda7a-def5-0000-0000-000000000005",
+	0,
+	'',
+	"89815f41-55ac-4195-9852-20c958dbe0b9",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("098487bf-d779-4876-99f7-538d95d25a8a",
+	"73a94ff8-3736-4fc5-b3ef-4d1b9823e200",
+	'checkLocation',
+	"ba5eda7a-def5-0000-0000-000000000001",
+	0,
+	'',
+	"d7f33178-5c0b-46b5-8969-f82b80b99e87",
+	'');
+INSERT INTO O_TFR
```

Comment:
does this operation make more sense on V_VAL?

Reply:
agreed, moved